Std/Classes/Algebra.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2023 Lean FRO. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joe Hendrix, Leonardo de Moura
+-/
+
+/-!
+This module reintroduces the algebraic class in
+@Mathlib.Init.Algebra.Classes@.  It is described in detail on [Github
+wiki](https://github.com/leanprover/lean/wiki/Refactoring-structures#encoding-the-algebraic-hierarchy-1)
+with pros and cons relevant to Lean 3.
+
+These classes index properties by the operation rather than the type as
+in Haskell.
+-/
+
+namespace Std
+
+/-- A symmetric operation. -/
+class IsSymmOp {α : Type u} {β : Sort v} (op : α → α → β) : Prop where
+  /-- Symmetric operation -/
+  symm_op : ∀ a b, op a b = op b a
```

Comment:
Where is this used other than the one derived instance below? I think this could stay in Mathlib until needed.

Reply:
It's not.  I added it because it was next to the other declarations and we have a `Symm` tactic.

Reply:
I experimented with deleting `ac_rfl` from core and nothing seems to break: https://github.com/fgdorais/lean4/tree/ac_rfl